### PR TITLE
Pkg 6116

### PR DESCRIPTION
**Pull request by Jrice1317**
libmamba, libmambapy 2.0.5

**Destination channel:** {defaults}

### Links

- [PKG-6116](https://anaconda.atlassian.net/browse/PKG-6116) 
- [Upstream repository](https://github.com/mamba-org/mamba)
- [Upstream changelog/diff](https://github.com/mamba-org/mamba/blob/micromamba-2.0.4/CHANGELOG.md)
- Relevant dependency PRs:
  - [`simdjson 3.10.1`](https://github.com/AnacondaRecipes/simdjson-feedstock/pull/1/)
  - [`libsolv 0.7.30`](https://github.com/AnacondaRecipes/libsolv-feedstock/pull/5)
  - [`cpp-expected 1.1.0`](https://github.com/AnacondaRecipes/cpp-expected-feedstock/pull/2)
  - [`mamba 2.0.4`](https://github.com/AnacondaRecipes/mamba-feedstock/pull/19)
 
### Explanation of changes:

-Credits To:
  - @cbouss 
  - @ELundby45 
  - Jamie Rodríguez-Guerra

[PKG-6116]: https://anaconda.atlassian.net/browse/PKG-6116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -69,7 +81,7 @@ outputs:
         {% set libmamba_version_split = libmamba_version.split('.') %}
         - cat $PREFIX/include/mamba/version.hpp | grep "LIBMAMBA_VERSION_MAJOR {{ libmamba_version_split[0] }}"  # [unix]
         - cat $PREFIX/include/mamba/version.hpp | grep "LIBMAMBA_VERSION_MINOR {{ libmamba_version_split[1] }}"  # [unix]
-        - cat $PREFIX/include/mamba/version.hpp | grep "LIBMAMBA_VERSION_PATCH {{ libmamba_version_split[2] }}"  # [unix]
+        # - cat $PREFIX/include/mamba/version.hpp | grep "LIBMAMBA_VERSION_PATCH {{ libmamba_version_split[2] }}"  # [unix]
```
Comment (cbouss) [inline recipe/meta.yaml]: Can this test be re-enabled?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -113,8 +130,9 @@ outputs:
       imports:
         - libmambapy
         - libmambapy.bindings
+        - libmambapy.solver
       commands:
-        - python -c "import libmambapy._version; assert libmambapy._version.__version__ == '{{ libmambapy_version }}'"
+        # - python -c "import libmambapy._version; assert libmambapy._version.__version__ == '{{ libmambapy_version }}'"
```
Comment (cbouss) [inline recipe/meta.yaml]: Can this test be re-enabled?

**Review by cbouss — COMMENTED** [review]
Could we align the build scripts with conda-forge?
Here we are building libmambapy twice.

File: recipe/meta.yaml (hunk 2 above)
Comment (Jrice1317) [reply recipe/meta.yaml]: I will try again but it was giving me some failures at one point.

File: recipe/meta.yaml (hunk 1 above)
Comment (Jrice1317) [reply recipe/meta.yaml]: I will try again but it was giving me some failures at one point.

File: recipe/meta.yaml (hunk 2 above)
Comment (Jrice1317) [reply recipe/meta.yaml]: There's no longer an underscore in front of version, so I fixed that and it works now! (:

File: recipe/meta.yaml (hunk 1 above)
Comment (Jrice1317) [reply recipe/meta.yaml]: Done!

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Comment by cbouss** [comment]
The build scripts still need to be corrected here.

**Comment by bwingconda** [comment]
You'll have to create that `build-lib` folder in the `build_mamba.bat` before you build in it. It's also spitting out `Skipping LTO for target libmamba::libmamba-dyn, the build type is not a release` which makes me think there should be a `CMAKE_BUILD_TYPE=Release` set on the cmake configure call.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,20 +1,28 @@
 {% set name = "mamba" %}
-{% set libmamba_version = "1.5.11" %}
-{% set libmambapy_version = "1.5.11" %}
-{% set mamba_version = "1.5.11" %}
-{% set release = "2024.11.22" %}
+{% set libmamba_version = "2.0.5" %}
+{% set libmambapy_version = "2.0.5" %}
+{% set mamba_version = "2.0.5" %}
```
Comment (bwingconda) [inline recipe/meta.yaml:4]: Since there is a single source, I think it's safe to have a single version number, right?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -113,8 +120,9 @@ outputs:
       imports:
         - libmambapy
         - libmambapy.bindings
+        - libmambapy.solver
       commands:
-        - python -c "import libmambapy._version; assert libmambapy._version.__version__ == '{{ libmambapy_version }}'"
+        - python -c "import libmambapy.version; assert libmambapy.version.__version__ == '{{ libmambapy_version }}'"
```
Comment (bwingconda) [inline recipe/meta.yaml:125]: Could run their unit tests for the python module
https://github.com/mamba-org/mamba/tree/main/libmambapy/tests

File: recipe/meta.yaml [hunk 5]
```diff
@@ -50,10 +57,18 @@ outputs:
...
         - reproc
+        - fmt
+        - spdlog
+        - cpp-expected
+        - nlohmann_json
+        - simdjson
+        - yaml-cpp
     test:
       commands:
         - test -d ${PREFIX}/include/mamba                                   # [unix]
```
Comment (bwingconda) [inline recipe/meta.yaml:74]: Likewise, there are tests for their C++ module as well.
https://github.com/mamba-org/mamba/tree/main/libmamba/tests

**Review by bwingconda — APPROVED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (Jrice1317) [reply recipe/meta.yaml:4]: I agree with your comments, but we're trying to stay close to conda-forge

**Pull request merged**